lib/fileSystem.js

```diff
@@ -0,0 +1,115 @@
+var q = require('q');
+
+/**
+ * @param {object} e error object
+ */
+function _fsErrorHandler(e) {
+  var msg = '';
+
+  switch (e.code) {
+  case FileError.QUOTA_EXCEEDED_ERR:
+    msg = 'QUOTA_EXCEEDED_ERR';
+    break;
+  case FileError.NOT_FOUND_ERR:
+    msg = 'NOT_FOUND_ERR';
+    break;
+  case FileError.SECURITY_ERR:
+    msg = 'SECURITY_ERR';
+    break;
+  case FileError.INVALID_MODIFICATION_ERR:
+    msg = 'INVALID_MODIFICATION_ERR';
+    break;
+  case FileError.INVALID_STATE_ERR:
+    msg = 'INVALID_STATE_ERR';
+    break;
+  default:
+    msg = 'Unknown Error';
+    break;
+  }
+
+  throw new Error('Error: ' + msg);
+}
+
+function _requestFileSystem() {
+  var d = q.defer();
+  navigator.webkitPersistentStorage.requestQuota(1024 * 1024, function(grantedBytes) {
```

Comment:
Best to use local storage adapter which would switch to be best adapter available on the device. Example from our JS SDK: https://github.com/feedhenry/fh-js-sdk/tree/master/libs/lawnchair